src/Servant/Reflex/Response.hs

```diff
@@ -0,0 +1,97 @@
+{-# LANGUAGE ConstraintKinds   #-}
+{-# LANGUAGE FlexibleContexts  #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+-- | Response handlers
+--   Usefull for 'clientWithOptsAndResultHandler'
+module Servant.Reflex.Response
+  ( reloadOnAPIError
+  , reloadOnResponse
+  , RefreshSupport
+  , isAuthErr
+  , is4xxErr
+  , isDecodeError
+  ) where
+
+import           Control.Monad               (join, void)
+import qualified Data.Text                   as Text
+import           JSDOM                       (currentDocument)
+import           JSDOM.Generated.Document    (getLocation)
+import           JSDOM.Types                 (Location, liftDOM)
+import           Language.Javascript.JSaddle (MonadJSM, ( # ))
+import           Reflex
+import           Reflex.Dom.Builder.Class
+import           Reflex.Dom.Widget.Basic
+import           Reflex.Dom.Xhr
+import           Servant.Reflex
+
+-- | Alias for constraints that can refresh
+type RefreshSupport t m
+   = (MonadJSM m, MonadHold t m, DomBuilder t m, SupportsServantReflex t m)
+
+is4xxErr :: ReqResult () a -> Bool
+is4xxErr =
+  maybe
+    False
+    (\req ->
+       (_xhrResponse_status req < fromInteger 500) &&
+       (_xhrResponse_status req >= 400)) .
+  response
+
+isAuthErr :: ReqResult () a -> Bool
+isAuthErr =
+  maybe
+    False
+    (\req ->
+       let stat = _xhrResponse_status req
+       in (stat == fromInteger 401) &&
+           (stat == fromInteger 402) && (stat == fromInteger 407)) .
+  response
+
+isDecodeError :: ReqResult () a -> Bool -- TODO find a better way of detecting this, is there no typesafe decode error?
+isDecodeError = maybe False (Text.isInfixOf "Error in $") . reqFailure
+
+-- | Reloads page and clears the cache on any 4xx response which is not auth related
+--   or on any decoding errors.
+--   Most endpoints want to have this.
```

Comment:
If most endpoints truly want to be wrapped in `reloadOnAPIError`, perhaps it would help to add a field to the global client options, ` , apiErrorBehavior :: APIErrorBehavior`. where `data APIErrorBehavior = Standard | Reload | Modal`. Then we could let servant-reflex manage the behavior globally, if the user wants that.

Reply:
Hmm, I'm unsure about that. I don't really see the direct benefits of using a sum type instead of just the monad approach.